src/equations/shallow_water_2d.jl

```diff
@@ -0,0 +1,619 @@
+  return max_abs_speed_naive(u_ll, u_rr, 0, equations) * norm(normal_direction)
+end
+
+
+# Calculate minimum and maximum wave speeds for HLL-type fluxes
+@inline function min_max_speed_naive(u_ll, u_rr, orientation::Integer,
+                                     equations::ShallowWaterEquations2D)
+  h_ll = u_ll[1]
+  _, v1_ll, v2_ll, _ = cons2prim(u_ll, equations)
+  h_rr = u_rr[1]
+  _, v1_rr, v2_rr, _ = cons2prim(u_rr, equations)
+
+  if orientation == 1 # x-direction
+    λ_min = v1_ll - sqrt(equations.gravity * h_ll)
+    λ_max = v1_rr + sqrt(equations.gravity * h_rr)
+  else # y-direction
+    λ_min = v2_ll - sqrt(equations.gravity * h_ll)
+    λ_max = v2_rr + sqrt(equations.gravity * h_rr)
+  end
+
+  return λ_min, λ_max
+end
+
+
+@inline function min_max_speed_naive(u_ll, u_rr, normal_direction::AbstractVector,
+                                     equations::ShallowWaterEquations2D)
+  h_ll = u_ll[1]
+  _, v1_ll, v2_ll, _ = cons2prim(u_ll, equations)
+  h_rr = u_rr[1]
+  _, v1_rr, v2_rr, _ = cons2prim(u_rr, equations)
+
+  v_normal_ll = v1_ll * normal_direction[1] + v2_ll * normal_direction[2]
+  v_normal_rr = v1_rr * normal_direction[1] + v2_rr * normal_direction[2]
+
+  norm_ = norm(normal_direction)
+  # The v_normals are already scaled by the norm
+  λ_min = v_normal_ll - sqrt(equations.gravity * h_ll) * norm_
+  λ_max = v_normal_rr + sqrt(equations.gravity * h_rr) * norm_
+
+  return λ_min, λ_max
+end
+
+
+@inline function max_abs_speeds(u, equations::ShallowWaterEquations2D)
+  h = u[1]
+  _, v1, v2, _ = cons2prim(u, equations)
+
+  c = equations.gravity * sqrt(h)
+  return abs(v1) + c, abs(v2) + c
+end
+
+
+# Convert conservative variables to primitive
+@inline function cons2prim(u, equations::ShallowWaterEquations2D)
+  h, h_v1, h_v2, b = u
+
+  H = h + b
+  v1 = h_v1 / h
+  v2 = h_v2 / h
+  return SVector(H, v1, v2, b)
+end
+
+
+# Convert conservative variables to entropy
+# Note, only the first three are the entropy variables, the fourth entry still
+# just carries the bottom topography values for convenience
+@inline function cons2entropy(u, equations::ShallowWaterEquations2D)
+  h, h_v1, h_v2, b = u
+
+  v1 = h_v1 / h
+  v2 = h_v2 / h
+  v_square = v1^2 + v2^2
+
+  w1 = equations.gravity * (h + b) - 0.5 * v_square
+  w2 = v1
+  w3 = v2
+  return SVector(w1, w2, w3, b)
+end
+
+
+# Convert entropy variables to conservative
+@inline function entropy2cons(w, equations::ShallowWaterEquations2D)
+  w1, w2, w3, b = w
+
+  h = (w1 + 0.5 * (w2^2 + w3^2)) / equations.gravity - b
+  h_v1 = h * w2
+  h_v2 = h * w3
+  return SVector(h, h_v1, h_v2, b)
+end
+
+
+# Convert primitive to conservative variables
+@inline function prim2cons(prim, equations::ShallowWaterEquations2D)
+  H, v1, v2, b = prim
+
+  h = H - b
+  h_v1 = h * v1
+  h_v2 = h * v2
+  return SVector(h, h_v1, h_v2, b)
+end
+
+
+@inline function water_height(u, equations::ShallowWaterEquations2D)
```

Comment:
```suggestion
@inline function waterheight(u, equations::ShallowWaterEquations2D)
```
you could consider renaming it such that it is clear that it is not "water times height". Yes, yes, the chance for confusion is low, but it would make it nicer if you combine it with another variable, e.g., `waterheight_density`. I don't know, maybe my suggestion is stupid :shrug:

Reply:
I am fine renaming this. Currently, the functions you mention are unused, but I kept them around because they would be needed for the curvilinear shock capturing / positivity preservation.